src/translib/transformer/xfmr_route_cmn.go

```diff
@@ -0,0 +1,146 @@
+package transformer
+
+import (
+    "errors"
+    "strings"
+    log "github.com/golang/glog"
+)
+
+
+func init () {
+    XlateFuncBind("YangToDb_route_table_conn_key_xfmr", YangToDb_route_table_conn_key_xfmr)
+    XlateFuncBind("DbToYang_route_table_conn_key_xfmr", DbToYang_route_table_conn_key_xfmr)
+    XlateFuncBind("YangToDb_route_table_addr_family_xfmr", YangToDb_route_table_addr_family_xfmr)
+    XlateFuncBind("DbToYang_route_table_addr_family_xfmr", DbToYang_route_table_addr_family_xfmr)
+}
+
+var YangToDb_route_table_addr_family_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+    res_map := make(map[string]string)
+
+    res_map["NULL"] = "NULL"
+    return res_map, nil
+}
+
+var DbToYang_route_table_addr_family_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+
+    var err error
+    result := make(map[string]interface{})
+
+    entry_key := inParams.key
+    key := strings.Split(entry_key, "|")
+    family := key[3]
+
+    result["address-family"] = family 
+
+    return result, err
+}
+
+
+var YangToDb_route_table_conn_key_xfmr KeyXfmrYangToDb = func(inParams XfmrParams) (string, error) {
+    var err error
+    log.Info("YangToDb_route_table_conn_key_xfmr***", inParams.uri)
+    pathInfo := NewPathInfo(inParams.uri)
+
+    niName     :=  pathInfo.Var("name")
+    srcProto   := pathInfo.Var("src-protocol")
+    dstProto   := pathInfo.Var("dst-protocol")
+    afName     := pathInfo.Var("address-family")
+
+    if len(pathInfo.Vars) < 3 {
+        err = errors.New("Invalid Key length");
+        log.Info("Invalid Key length", len(pathInfo.Vars))
+        return niName, err
+    }
+
+    if len(niName) == 0 {
+        err = errors.New("vrf name is missing");
+        log.Info("VRF Name is Missing")
+        return niName, err
+    }
+
+    var family string
+    var source string
+    var destination string
+
+    if strings.Contains(afName, "IPV4") {
+        family = "ipv4"
+    } else if strings.Contains(afName, "IPV6") {
+        family = "ipv6"
+    } else {
+		log.Info("Unsupported address-family " + afName)
+		return family, errors.New("Unsupported address-family " + afName)
+    }
+
+    if strings.Contains(srcProto, "DIRECTLY_CONNECTED") {
```

Comment:
Use switch case here. Else the last else will have to go through all the string comparisons.